src/jit/sideeffects.cpp

```diff
@@ -450,7 +450,7 @@ void SideEffectSet::AddNode(Compiler* compiler, GenTree* node)
 //    Two side effect sets interfere under any of the following
 //    conditions:
 //    - If the analysis is strict, and:
-//        - Either set contains a compiler barrier, or
+//        - One set contains a compiler barrier and the other set contains a global reference, or
```

Comment:
I believe that when both sides are marked with GTF_ORDER_SIDEEFF you also cannot reorder.

So you should add a check for that as well.

Reply:
It is quite possible that all indirections that set GTF_ORDER_SIDEEFF also will set GTF_GLOB_REF